### PR TITLE
Replace platform conditionals in whole remediation code

### PR DESCRIPTION
**Pull request by jan-cerny**
The XCCDF `<fix>` elements can contain XCCDF `<sub>` elements,
therefore to perform a replacement in the whole remediation code
we need to replace strings not only in the `<fix>` element's text
but also in `<sub>` element`s tail.

Not doing so caused that the platform conditionals that occurred
after `<sub>` elements were not replaced. As a result, some
remediations were skipped when using a container back end, which
caused the test runs to fail.

### CODE REVIEW

**Comment by marcusburghardt** [comment]
Hi @jan-cerny , what is your suggestion about the best way to test this improvement? Can you provide an example, please?

**Comment by jan-cerny** [comment]
@marcusburghardt 

You need to run test scenarios for some rule that 1. has Ansible remeditation, 2. is not applicable to containers 3. is parametrized by a variable.

I used the rule `sshd_set_max_auth_tries` in a RHEL 8 datastream, because I noticed this issue during a review of #9344 which updates the `sshd_set_max_auth_tries`.  For example:
```
$ tests/test_rule_in_container.sh --dontclean --remediate-using ansible sshd_set_max_auth_tries 
```
In the test run logs directory, check the generated Ansible remediation, in this example open the `xccdf_org.ssgproject.content_rule_sshd_set_max_auth_tries.yml` and watch then `when:` statement whether it contains `when: True`.

Also, you can check the ARF files (eg. `sshd_set_max_auth_tries-line_not_there.fail.sh-initial-arf.xml`) in the test run logs directory to make sure they don't contain `ansible_virtualization_type not in ["docker", "lxc", "openvz", "podman", "container"]`.

For the Bash remediations, it's similar, watch for `if true ; then`, but unfortunately I don't know from the top of my head about a Bash remediation where this issue occurs, because usually the `if [ -f /.containerev ] ` conditions appear before the `<sub>` XCCDF element.

**Comment by jan-cerny** [comment]
@marcusburghardt  I just realized that you might miss the context. So the context is: We can test rules either on a virtual machine back end or on a container back end. However, a lot of rules are marked as machine only platform. These rules are not useful in containers for a general container user but they are testable in containers. Normally, these rules would be skipped and their remediations would be skipped when run on container back end. To avoid skipping them, AutoMatúš has some mechanisms that modify the tested SCAP data stream before the tests are executed.  It removes the platform limitations and platform conditions from these rules to make them run on the container back ends.

**Comment by jan-cerny** [comment]
I have rebased this PR on the top of the latest upstream master branch because they have changed the required set of tests in our GitHub CI.

File: tests/ssg_test_suite/xml_operations.py [hunk 1]
```diff
@@ -122,27 +122,32 @@ def remove_machine_remediation_condition(root):
...
+    system = "urn:xccdf:fix:script:sh"
     considered_machine_platform_checks = [
         r"\[\s+!\s+-f\s+/\.dockerenv\s+\]\s+&&\s+\[\s+!\s+-f\s+/run/\.containerenv\s+\]",
     ]
-    for el in fix_elements:
-        if not el.text:
-            continue
-        for check in considered_machine_platform_checks:
-            el.text = re.sub(check, "true", el.text)
+    repl = "true"
```
Comment (marcusburghardt) [inline tests/ssg_test_suite/xml_operations.py:129]: It is not consistent with the line [138](https://github.com/ComplianceAsCode/content/pull/9347/files#diff-414bcf1a7602826f6b3856ff50d4d9809457bd22a7a332e6362df404d749cca1R138).

**Comment by marcusburghardt** [comment]
> @marcusburghardt I just realized that you might miss the context. So the context is: We can test rules either on a virtual machine back end or on a container back end. However, a lot of rules are marked as machine only platform. These rules are not useful in containers for a general container user but they are testable in containers. Normally, these rules would be skipped and their remediations would be skipped when run on container back end. To avoid skipping them, AutoMatúš has some mechanisms that modify the tested SCAP data stream before the tests are executed. It removes the platform limitations and platform conditions from these rules to make them run on the container back ends.

Thanks for the context @jan-cerny. I will test the PR right now.

**Comment by marcusburghardt** [comment]
Before the patch, indeed the conditional was not changed during the tests:
```
    - name: Set SSH authentication attempt limit
      block:

      - name: Check for duplicate values
        lineinfile:
        ...
      when: ansible_virtualization_type not in ["docker", "lxc", "openvz", "podman", "container"]
```
After the patch, the conditional was updated as expected:
```
    - name: Set SSH authentication attempt limit
      block:

      - name: Check for duplicate values
        lineinfile:
        ...
      when: True
```

File: tests/ssg_test_suite/xml_operations.py (hunk 1 above)
Comment (marcusburghardt) [reply tests/ssg_test_suite/xml_operations.py:129]: Actually, this is fine for the bash equivalent function. Never mind!

**Review by marcusburghardt — APPROVED** [review]
Thanks @jan-cerny

**Pull request merged**